python/nano/src/bigdl/nano/tf/utils.py

```diff
@@ -13,7 +13,70 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import inspect
 import operator
+from tensorflow.keras.layers import Layer
+from functools import partial
 from bigdl.nano.common.compare_version import _compare_version
 
 KERAS_VERSION_LESS_2_9 = _compare_version("keras", operator.lt, "2.9")
+
+
+class _NanoPartial(partial):
+    pass
+
+
+def _create_wrapper_type(parent_type, target_obj, source_obj):
+    def _getattr(self, name):
+        try:
+            return getattr(self.target_obj, name)
+        except AttributeError as _e:
+            pass
+        return getattr(self.source_obj, name)
+
+    def _setattr(self, name, value):
+        return setattr(self.target_obj, name, value)
+
+    def _call(self, *args, **kwargs):
+        return self.target_obj(*args, **kwargs)
+
+    return type(
+        f"wrapped_{parent_type.__class__.__name__}", (parent_type,),
+        {
+            'target_obj': target_obj,
+            'source_obj': source_obj,
+            '__getattr__': _getattr,
+            '__setattr__': _setattr,
+            '__call__': _call,
+        }
+    )
+
+
+def patch_attrs(target_obj: object, source_obj: object) -> object:
+    """
+    Patch attributes of `source_obj` to `target_obj`.
+
+    :param target_obj: target object
+    :param source_obj: source object
+    :return: `target_obj`
+    """
+    if inspect.ismethod(target_obj.__setattr__):
```

Comment:
maybe this one will cover all the cases, since all obj has `__setattr__` right?

Reply:
no, inspect.ismethod('__setattr__') is `True` only when `__setattr__` is defined explicitly by user

Reply:
Yeah, I had a try you are right